src/components/TableWrapper.tsx

```diff
@@ -0,0 +1,52 @@
+import { Button, Table } from 'antd';
+import { ColumnsType } from 'antd/lib/table';
+import React from 'react';
+
+export default function TableWrapper<RecordType extends object>({
+  dataSource,
+  columns,
+  customAction,
+  customTitle,
+}: {
+  dataSource: RecordType[];
```

Comment:
I recommend move definition of props into separate file. It's hard to read code now.